From 09f8afdac677cbeb12d528dafc3e663a80414076 Mon Sep 17 00:00:00 2001
From: Gu, Wangyi <wangyi.gu@intel.com>
Date: Tue, 5 Feb 2013 11:41:08 +0800
Subject: Movie studio: fix the crash issue when import an especial clip

BZ: 81210

root cause: the first sync frame(timestamp=0) can't be found in the sync sample
box, when we need seek to 0s with the seek mode "SEEK_PREVIOUS_SYNC", will crash
in the CHECK expression, it's android defect, we return the first sync frame while
this case happen.

Category:aosp improvement
Domain: Video.Media-video editor
Origin: internal
Upstream-Candidate: Yes

Change-Id: Ic02ad3b82bf9893c7e3a44f53bdbd2e000e78368
Orig-Change-Id: I41c1d3a573fc7bc496116eb085944255d058e2af
Signed-off-by: Gu, Wangyi <wangyi.gu@intel.com>
Signed-off-by: Dan Liang <dan.liang@intel.com>
Signed-off-by: chuzhoux <chux.zhou@intel.com>
---
 media/libstagefright/SampleTable.cpp |   18 +++++++++++-------
 1 files changed, 11 insertions(+), 7 deletions(-)

diff --git a/media/libstagefright/SampleTable.cpp b/media/libstagefright/SampleTable.cpp
index 75f63d5..db3a008 100644
--- a/media/libstagefright/SampleTable.cpp
+++ b/media/libstagefright/SampleTable.cpp
@@ -692,14 +692,18 @@ status_t SampleTable::findSyncSampleNear(
         case kFlagBefore:
         {
             if (x > start_sample_index) {
-                CHECK(left > 0);
 
-                x = mSyncSamples[left - 1];
-
-                if (x > start_sample_index) {
-                    // The table of sync sample indices was not sorted
-                    // properly.
-                    return ERROR_MALFORMED;
+                if (left > 0) {
+                    x = mSyncSamples[left - 1];
+
+                    if (x > start_sample_index) {
+                        // The table of sync sample indices was not sorted
+                        // properly.
+                        return ERROR_MALFORMED;
+                    }
+                } else {
+                    // If seek time is 0, return the first sample as a sync sample
+                    x = 0;
                 }
             }
             break;
-- 
1.7.4.1

